document-store/src/main/java/org/hypertrace/core/documentstore/query/GenericOrderBy.java

```diff
@@ -0,0 +1,14 @@
+package org.hypertrace.core.documentstore.query;
+
+import lombok.Value;
+import org.hypertrace.core.documentstore.expression.Sortable;
+
+/**
+ * A generic ORDER BY definition that supports expressions. Note that this class is a more general
+ * version of {@link org.hypertrace.core.documentstore.OrderBy}
+ */
+@Value(staticConstructor = "of")
+public class GenericOrderBy {
+  Sortable sortable;
+  boolean ascending;
```

Comment:
personal nit: I've always found using a boolean to represent this weird, but may just be me. I just find something like `setDirection(DESC)` compared to `setAscending(false)` to be more readable.

Reply:
Makes sense. I also prefer this better.